File: C/0205-Isomorphic-Strings.cpp
    bool isIsomorphic(string s, string t) {
        unordered_map<char, char> hashmap;
        unordered_set<char> hashset;
        
        for (int i = 0; i < s.length(); ++i) {
            if (hashmap.find(s[i]) == hashmap.end() && hashset.find(t[i]) == hashset.end()) {
                hashmap[s[i]] = t[i];
                hashset.insert(t[i]);
            } else if (hashmap[s[i]] != t[i]) {
                return false;
            }
        }
        
        return true;
    }

File: Python/0205-Isomorphic-Strings.py
    def isIsomorphic(self, s, t):
        d = {}
        
        for (a, b) in zip(s, t):
            if a not in d.keys():
                if b not in d.values():
                    d[a] = b
                else:
                    return False
            elif d[a] != b:
                return False
        return True
